stock_available_sale/models/product_product.py

```diff
@@ -0,0 +1,118 @@
+        # when we search locations, should children be searched too?
+        if self.env.context.get('compute_child', True):
+            loc_op = 'child_of'
+        else:
+            loc_op = 'in'
+        # Limit to some locations
+        # Take warehouses that have these locations as stock locations
+        if self.env.context.get('location', False):
+            # Search by ID
+            if isinstance(self.env.context['location'], (int, long)):
+                domain.append(
+                    ('order_id.warehouse_id.lot_stock_id', loc_op,
+                     [self.env.context['location']]))
+            # Search by name
+            elif isinstance(self.env.context['location'], basestring):
+                location_ids = [
+                    l.id
+                    for l in self.env['stock.location'].search([
+                        ('complete_name', 'ilike',
+                         self.env.context['location'])])]
+                domain.append(
+                    ('order_id.warehouse_id.lot_stock_id', loc_op,
+                     location_ids))
+            # Search by whatever the context has - probably a list of IDs
+            else:
+                domain.append(
+                    ('order_id.warehouse_id.lot_stock_id', loc_op,
+                     self.env.context['location']))
+        # Limit to a warehouse
+        if self.env.context.get('warehouse', False):
+            domain.append(
+                ('order_id.warehouse_id', '=', self.env.context['warehouse']))
+        # Limit to a period
+        from_date = self.env.context.get('from_date', False)
+        to_date = self.env.context.get('to_date', False)
+        if from_date:
+            domain.extend([
+                ('order_id.requested_date', '>=', from_date),
+                '&',  # only consider 'date' when 'equested_date' is empty
+                ('order_id.requested_date', '=', False),
+                ('order_id.date', '>=', from_date),
+                ])
+        if to_date:
+            domain.extend([
+                ('order_id.requested_date', '<=', to_date),
+                '&',  # only consider 'date' when 'equested_date' is empty
+                ('order_id.requested_date', '=', False),
+                ('order_id.date', '<=', to_date),
+                ])
+
+        # Compute the quoted quantity for each product
+        results = Counter()
+        for group in self.env['sale.order.line'].read_group(
+                domain, ['product_uom_qty', 'product_id', 'product_uom'],
+                ['product_id', 'product_uom'],
+                lazy=False):
+            product_id = group['product_id'][0]
+            uom_id = group['product_uom'][0]
+            # Compute the quoted quantity in the product's UoM
+            # Rounding is OK since small values have not been squashed before
+            results += Counter({
+                product_id: self.env['product.uom'].sudo()._compute_qty(
```

Comment:
@lmignon I'm doing a `sudo` also in `stock_available_sale`, do you think it could break multi-company setups like it did in `stock_available_mrp`?